### PR TITLE
Beautify tests

### PR DESCRIPTION
**Pull request by dns2utf8**
With love

### CODE REVIEW

File: src/user_input_compiler.rs [hunk 1]
```diff
@@ -825,14 +825,7 @@ mod test {
...
     match Input::from( format!("/webcam 42") ) {
-      WebCam{ nth } => {
-        if let Some(nth) = nth { 
-          assert!( if nth == 42 { true } else { println!("wrong Value: {}", nth); false } ) 
-        } else {
-          println!("expected OptionalInteger");
-          assert!(false)
-        }
-      },
+      WebCam{ nth } => assert_eq!(Some(42), nth),
```
Comment (rnestler) [inline src/user_input_compiler.rs]: I think you could reduce it even further to `assert_eq!(WebCam{ nth=Some(42)}, Input::from(...))`. See [here](https://play.rust-lang.org/?code=%0A%23%5Bderive%28Debug%2C%20PartialEq%2C%20Eq%29%5D%0Astruct%20Foo%20%7B%0A%20%20%20%20i%3A%20i32%2C%0A%7D%0A%0Afn%20main%28%29%20%7B%0A%20%20%20%20let%20foo%20%3D%20Foo%7Bi%3A%2010%7D%3B%0A%20%20%20%20assert_eq!%28foo%2C%20Foo%7Bi%3A%2010%7D%29%3B%0A%7D%0A&version=stable).

File: src/user_input_compiler.rs [hunk 2]
```diff
@@ -794,57 +794,33 @@ mod test {
...
   #[test]
   fn webcam_negative_13() {
-    let r = Input::from( format!("/webcam -13") );
-    
-    if let InvalidSyntax(_) = r {
-    } else {
-        println!("Unexpected: {:?}", r);
-        assert!(false) ;
-    }
+    assert_eq!( InvalidSyntax("Expected positive Integer".into()), Input::from( format!("/webcam -13") ))
```
Comment (rnestler) [inline src/user_input_compiler.rs:630]: Hmm... The test for the exact error message is maybe a bit too much, but I would leave it for now until it gets annoying to fix the unit tests.

**Comment by rnestler** [comment]
:+1:

**Comment by dns2utf8** [comment]
I applied the beautification to all files.

Reduced CodeBase by 220 lines :grin:

File: src/grammar.rs [hunk 3]
```diff
@@ -43,7 +43,7 @@ mod test {
   #[test]
   fn print_grammar_string() {
     let g :String = get_grammar_string();
-    info!("alles: {:?}", g);
+    println!("alles: {:?}", g);
```
Comment (rnestler) [inline src/grammar.rs:46]: Whats wrong with using logging?

Comment (dns2utf8) [inline src/grammar.rs:46]: It is not visible when the Test fails. It worked with rust 1.5

Comment (dbrgn) [inline src/grammar.rs:46]: Are you initializing a logger in the test code? When using env_logger, I think you need to initialize it in every test run, as they're run in parallel.

Comment (dns2utf8) [inline src/grammar.rs:46]: This seems very complicated, I leave it with `println!` for now, since `cargo` hides the output from successful tests.

Comment (dbrgn) [inline src/grammar.rs:46]: Yep, I think println is fine in tests.

**Pull request merged**